confgenerator/confgenerator.go

```diff
@@ -102,20 +103,58 @@ func (m *Metrics) generateOtelPipelines() (map[string]otel.Pipeline, error) {
 	return out, nil
 }
 
+type sortKey struct {
+	n   int
+	tag string
+}
+
+func sortKeyLess(a, b sortKey) bool {
+	return a.n < b.n || (a.n == b.n && a.tag < b.tag)
+}
+
+func inputSortKey(i fluentbit.Input) sortKey {
+	switch r := i.(type) {
+	case *fluentbit.Tail:
```

Comment:
Curious: Do we need the `1`, `2`, `3` based ordering here because otherwise the result is non-deterministic or because we have to enforce the exact orders of these configs? For some reason, we can not just rely on `tail`, `syslog` and `windows_event_log` to be sorted alphabetically?

Can we add a comment for the rationale?

Reply:
I'm leaning towards the receivers / processors being sorted alphabetically in the long run, so that we do not need to manually maintain this sorting order.

Reply:
Sync'd offline, the sorting here is just to minimize the diff in the golden files. The actual sorting will not be depending on the types and will be handled in a follow up PR.

Reply:
I believe the final sort order will be that pipelines will be generated alphabetically and then within each pipeline the filters will be generated in the user-specified order. I don't particularly care about this sorting code because it's just going to be removed soon anyway.